Library/Homebrew/livecheck/livecheck.rb

```diff
@@ -396,40 +427,44 @@ def formula_name(formula, full_name: false)
 
     sig {
       params(
-        formula_or_cask: T.any(Formula, Cask::Cask),
-        status_str:      String,
-        messages:        T.nilable(T::Array[String]),
-        full_name:       T::Boolean,
-        verbose:         T::Boolean,
+        package_or_resource: T.any(Formula, Cask::Cask, Resource),
+        status_str:          String,
+        messages:            T.nilable(T::Array[String]),
+        full_name:           T::Boolean,
+        verbose:             T::Boolean,
       ).returns(Hash)
     }
-    def status_hash(formula_or_cask, status_str, messages = nil, full_name: false, verbose: false)
-      formula = formula_or_cask if formula_or_cask.is_a?(Formula)
-      cask = formula_or_cask if formula_or_cask.is_a?(Cask::Cask)
+    def status_hash(package_or_resource, status_str, messages = nil, full_name: false, verbose: false)
+      formula = package_or_resource if package_or_resource.is_a?(Formula)
+      cask = package_or_resource if package_or_resource.is_a?(Cask::Cask)
+      resource = package_or_resource if package_or_resource.is_a?(Resource)
 
       status_hash = {}
       if formula
         status_hash[:formula] = formula_name(formula, full_name: full_name)
       elsif cask
-        status_hash[:cask] = cask_name(formula_or_cask, full_name: full_name)
+        status_hash[:cask] = cask_name(cask, full_name: full_name)
+      elsif resource
+        status_hash[:resource] = resource.name
       end
       status_hash[:status] = status_str
       status_hash[:messages] = messages if messages.is_a?(Array)
 
       status_hash[:meta] = {
-        livecheckable: formula_or_cask.livecheckable?,
+        livecheckable: package_or_resource.livecheckable?,
       }
       status_hash[:meta][:head_only] = true if formula&.head_only?
 
       status_hash
     end
 
-    # Formats and prints the livecheck result for a formula.
-    sig { params(info: Hash, verbose: T::Boolean, ambiguous_cask: T::Boolean).void }
-    def print_latest_version(info, verbose:, ambiguous_cask: false)
-      formula_or_cask_s = "#{Tty.blue}#{info[:formula] || info[:cask]}#{Tty.reset}"
-      formula_or_cask_s += " (cask)" if ambiguous_cask
-      formula_or_cask_s += " (guessed)" if !info[:meta][:livecheckable] && verbose
+    # Formats and prints the livecheck result for a formula/cask/resource.
+    sig { params(info: Hash, verbose: T::Boolean, ambiguous_cask: T::Boolean, resource: T::Boolean).void }
+    def print_latest_version(info, verbose:, ambiguous_cask: false, resource: false)
+      package_or_resource_s = resource ? "  " : ""
+      package_or_resource_s += "#{Tty.blue}#{info[:formula] || info[:cask] || info[:resource]}#{Tty.reset}"
+      package_or_resource_s += " (cask)" if ambiguous_cask
+      package_or_resource_s += " (guessed)" if verbose && !resource && !info[:meta][:livecheckable]
```

Comment:
```suggestion
      package_or_resource_s += " (guessed)" if verbose && !info[:meta][:livecheckable]
```

The behavior for resources should be the same as formulae in terms of the `(guessed)` output. That said, this suggested change will error because the `#resource_version` method isn't providing all the data we need in its return value. Basically, `#resource_version` should always return all of the data (since `#print_latest_version` needs `:meta` information) and `#run_checks` should take care of removing `:meta` from the resource hashes when the `--json` flag is set but the `--verbose` flag is not.

You'll need to remove a few early returns in `#resource_version` that are artificially restricting the returned data and add additional logic in `#run_checks` that mimics the existing `info.except!(:meta) unless verbose` logic (but applying to each resource hash in the resources array, of course).

If this isn't clear, take a look at the return value from `#latest_version` and how it's handled in `#run_checks`.

Reply:
Make sense. Done ✅